examples/text_classification/tc_mnli_transformers.ipynb

```diff
@@ -518,11 +604,17 @@
     "        to_lower=model_name.endswith(\"uncased\"),\n",
     "        cache_dir=CACHE_DIR,\n",
     "    )\n",
-    "    train_dataloader = processor.create_dataloader_from_df(\n",
-    "        df_train, TEXT_COL, LABEL_COL, max_len=MAX_LEN, batch_size=BATCH_SIZE, num_gpus=NUM_GPUS, shuffle=True\n",
+    "    train_dataset = processor.dataset_from_dataframe(\n",
+    "        df_train, TEXT_COL, LABEL_COL, max_len=MAX_LEN\n",
     "    )\n",
-    "    test_dataloader = processor.create_dataloader_from_df(\n",
-    "        df_test, TEXT_COL, LABEL_COL, max_len=MAX_LEN, batch_size=BATCH_SIZE, num_gpus=NUM_GPUS, shuffle=False\n",
+    "    train_dataloader = dataloader_from_dataset(\n",
```

Comment:
can we decide what data loader is needed in the fit and predict function so that we don't need to repeat the logic of handling distributed training in both data loader function and the fit function? We can just pass in the dataset for both fit and predict function.